sdk/messaging/azeventgrid/publisher/autorest.md

```diff
@@ -0,0 +1,68 @@
+## Go
+
+``` yaml
+title: EventGridPublisherClient
+description: Azure Event Grid client
+generated-metadata: false
+clear-output-folder: false
+go: true
+input-file: 
+    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/eventgrid/data-plane/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json
+license-header: MICROSOFT_MIT_NO_VERSION
+openapi-type: "data-plane"
+output-folder: ../publisher
+override-client-name: Client
+security: "AADToken"
+use: "@autorest/go@4.0.0-preview.52"
+version: "^3.0.0"
+slice-elements-byval: true
+remove-non-reference-schema: true
```

Comment:
Does `rawjson-as-bytes: true` need to be enabled as well or is that not applicable here?

Reply:
I don't think it's the right thing because the underlying []byte slice is either the raw bytes _or_ JSON bytes, depending on what they send. 

I did notice that we're generating it in a way that I believe we don't want to do, however. I filed a bug for it : https://github.com/Azure/autorest.go/issues/1022

Reply:
Reopening for a second. EventGrid doesn't do the fancy encoding-switch that CloudEvent does, which means we don't end up in a situation where the underlying value might just be a raw stream of bytes or JSON. It's always just data encoded into the body of the message.

I don't think that changes what we have today - I think it's preferable to allow the customer to just use an object or primitive here and let JSON encoding do the work when we send it, rather than relying on them serializing the data to []bytes first. This is one area where I think C#/Java's BinaryData makes this representation easier. 

Also, of note - we don't deserialize the Data in this SDK - there's an entire constellation of "System Events" that will do it but we don't yet have that as part of this package. We might want a different approach there, depending on how that works.